### DISCUSSION TITLE
If the chance don't proc the conversion the rule gets skipped

### DISCUSSION THREAD

**Discussion by Inferdy** [mxgmn/MarkovJunior#35] [General]
I have a parrallel rulenode with atleast 1 match but it has a probability of 0.25 and if all the matches don't proc the conversion the rule is skipped. Sometimes it may be useful, but not in my case. There must be a way to choose desired outcome.

**Comment by mxgmn** [comment]
I don't understand what you mean here. Elaborate please or formulate it better. What is conversion?

**Reply by Inferdy** [reply]
In other words the conversion is a replacement.
If you set a chance for the replacement and **there are correct matches** but **no replacements due to the chance** the rule **will be skipped** like there was no matches at all. It may be a desired outcome, but sometimes it isn't. So there must be a way to choose desired outcome.

**Reply by mxgmn** [reply]
Could you give an example when this is not the desired outcome?

**Reply by Inferdy** [reply]
For a model like this
```xml
<prl values="AIW" origin="Center" in="*I*" out="IWI" p="0.25"/>
```
the modelling will stop without any changes with a chance of 75%

**Reply by Inferdy** [reply]
So there will be just 2 equal frames with 1 point in the center.
![0](https://user-images.githubusercontent.com/24939153/174111403-9648b551-2d66-484b-90f8-be790011b1ff.png)
But if you are lucky enough, the process will continue
![6jvycp](https://user-images.githubusercontent.com/24939153/174112095-f2dd68a7-fd78-43a6-b17e-1b540bc611ff.gif)

**Reply by mxgmn** [reply]
What do you want to achieve by adding a probability here? Make the animation go slower?

**Reply by Inferdy** [reply]
I want to add some roughness. If you remove the probabilty it becomes boring:
![6juqox](https://user-images.githubusercontent.com/24939153/174113069-4c4cffa5-db09-411d-9d59-52c9ddcdf3f4.gif)

**Reply by Inferdy** [reply]
Just to note, it may stop after the first division aswell with one of these final results:
![1](https://user-images.githubusercontent.com/24939153/174115254-e3f0f1d9-c253-43be-b974-dc2204ac5e16.png)
![1](https://user-images.githubusercontent.com/24939153/174115452-18bf55e0-6dd3-48e8-861f-cde1aa9f5e41.png)
![1](https://user-images.githubusercontent.com/24939153/174115310-26aaa58a-2f75-40fa-a0ff-89e3e271e54f.png)
Here it stopped after the 2nd division:
![2](https://user-images.githubusercontent.com/24939153/174115569-c0cd9511-7177-4b0a-a05e-0a87bb6074f1.png)

**Reply by mxgmn** [reply]
It's boring only with the `prl` node. Use `all` to make it rough, use `one` to make it even rougher.

**Reply by Inferdy** [reply]
Well, the `prl` approach has some positive sides:
1) It's simple
2) It's quick
3) It's different (unique)
